### DISCUSSION TITLE
Tracing improvements + Performance upgrade

### DISCUSSION THREAD

**Discussion by marcuskohlberg** [encoredev/encore#513] [General]
## Tracing is now sooo much better

<img width="1632" alt="improved tracing view" src="https://user-images.githubusercontent.com/78424526/200013818-d062f072-c837-492c-b654-a9f13b486a28.png">

We've just released a much-improved trace view. Here's what you need to know:

* Raw endpoints are now traced – including HTTP path, method, request/response bodies and headers.
* Traces now come with a JSON viewer, letting you expand/collapse parts of the payload if it's unwieldy to view.
* Remember to update before trying it out: `encore version update`

## Speed improvements to parser and tests

To make Encore better for everyone with a very large code base, we've improved Encore's static analysis performance. The parser is now **5x** faster!

With Encore you run tests with `encore test` instead of `go test`. This is a wrapper that compiles the Encore app and then runs `go test`. It supports all the same flags that the `go test` command does. For bigger projects, we want to make sure this is snappy! – In the latest release, we've improved test speed **2x**, both for local development as well as for CI/CD

## Bugfixes and other improvements

We're always thankful when we get feedback and bug reports from anyone using Encore. Here are some improvements we've made lately, thanks to your input:

* The metadata APIs that provide information about the current request also contains information about PubSub messages and Cron Job executions.
* You can now override auth information for the current request inside tests.
* Improved documentation for [how to structure your Encore app](https://encore.dev/docs/develop/app-structure) (thanks to **@Minivera**!).

Thanks to our contributors
We rely on the community to improve Encore, and we continue to be surprised and humbled by your incredible support and encouragement. **A massive thank you to @Minivera , Patryk Siemiński, Amr, Daniel Stocks, @Willyham, MaxD, Vilhelm Melkstam, Jakob Nordfeldt, Prasanna Balaji, and Juan Álvarez for your contributions and feedback!**

## What’s next
We're close to releasing a big upgrade to infrastructure provisioning, making it more flexible and easier for you to add specific requirements like deploying to existing Kubernetes clusters. We'll also soon release new observability and metrics capabilities.
– Remember to vote on your favorite feature on the [roadmap](https://encore.dev/roadmap)!

As always, if you have questions or feedback, tell us on [Slack](https://encore.dev/slack) or post on the [Community Forum](https://community.encore.dev/).

We’re excited to hear your feedback! ❤️

Catch you in the cloud,
André & the Encore team

**Full Changelog**: https://github.com/encoredev/encore/compare/v1.9.3...v1.10.1

<hr /><em>This discussion was created from the release <a href='https://github.com/encoredev/encore/releases/tag/v1.10.1'>Tracing improvements + Performance upgrade</a>.</em>

**Comment by cbsmith402** [comment]
> You can now override auth information for the current request inside tests.

Are there docs for this?

**Reply by eandre** [reply]
Yes, in the package docs: https://pkg.go.dev/encore.dev/et#OverrideAuthInfo
